### PR TITLE
feat: add `@Attribute` & `@Table` decorators

### PR DESCRIPTION
**Pull request by ephys**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

This PR implements `@Table` & `@Attribute` from sequelize-typescript (see #15334). `@Column` has been included as well but is deprecated and renamed to `@Attribute`.

This PR also introduces the ability to load models through the Sequelize constructor:

```ts
new Sequelize({
  models: [User, Projects],
});
```

As well as through the `sequelize.addModels` method.

```ts
const sequelize = new Sequelize();

sequelize.addModels([User, Projects]);
```

This is necessary because decorated models are only initialized once they are attached to Sequelize using either solution.

Unlike sequelize-typescript, these two options/methods are unable to load models from the filesystem themselves. `importModels` must be used instead. This is because `importModels` is async, as it uses `import` instead of `require` to be ESM-ready.

```ts
import { importModels, Sequelize } from '@sequelize/core';

new Sequelize({
  models: await importModels([__dirname + '/**/*.ts'])
});
```

I've only imported the tests from sequelize-typescript that weren't redundant with our existing tests.

### CODE REVIEW

File: src/sequelize.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
 import isPlainObject from 'lodash/isPlainObject';
+import { initDecoratedModel } from './decorators/shared/model';
```
Comment (WikiRik) [inline src/sequelize.js:4]: Don't think this is used, right?

**Review by WikiRik — APPROVED** [review]
Just one small comment

File: test/unit/decorators/table.test.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+    function two() {}
+
+    @Table({ getterMethods: { one } })
+    @Table({ getterMethods: { two } })
+    class User extends Model {}
+
+    sequelize.addModels([User]);
+
+    expect(User.options.getterMethods).to.deep.equal({ one, two });
```
Comment (fzn0x) [inline test/unit/decorators/table.test.ts:162]: I think `setterMethods` and `getterMethods` will be deprecated, right?

Comment (fzn0x) [reply test/unit/decorators/table.test.ts:162]: Is it different or the same as https://sequelize.org/docs/v6/core-concepts/getters-setters-virtuals/#deprecated-in-sequelize-v7-gettermethods-and-settermethods? I think we must not include it in v7 tests then

Comment (ephys) [reply test/unit/decorators/table.test.ts:162]: Even if the feature is deprecated, it's currently still supported, so we need to test it. We'll remove this test once the feature is removed

Comment (fzn0x) [reply test/unit/decorators/table.test.ts:162]: > Even if the feature is deprecated, it's currently still supported, so we need to test it. We'll remove this test once the feature is removed

That's understandable :)

File: src/sequelize.js (hunk 1 above)
Comment (ephys) [reply src/sequelize.js]: can't wait until the whole codebase is written in TS so our linter catches this

Comment (ephys) [reply src/sequelize.js]: Done

**Review by WikiRik — APPROVED** [review]

**Review by fzn0x — APPROVED** [review]

**Pull request merged**